src/components/sidebar/Stats.jsx

```diff
@@ -10,59 +10,110 @@ import SparksIcon from '../svgIcons/societyIcons/Sparks';
+   *
+   * @property {string} title - Society name
+   * @property {string} page - page type
+   * @property {Object[]} stats - Statistics to display
+   */
+  static propTypes = {
+    title: PropTypes.string,
+    page: PropTypes.string,
+    stats: PropTypes.arrayOf(PropTypes.shape({
+      name: PropTypes.string.isRequired,
+      value: PropTypes.string.isRequired,
+    })).isRequired,
   };
 
-  const SocietyIcon = icons[props.title.toLowerCase()];
+  /**
+   * @name defaultProps
+   * @property {string} title - Society name
+   * @property {string} page - page type
+   */
+  static defaultProps = {
+    title: '',
+    page: '',
+  };
 
-  return (
-    <section className='stats'>
-      {
-        props.page === 'society' ?
-          <span className='sidebar__navIcon sidebar__navIcon--black'>
-            <SocietyIcon />
-          </span>
-          :
-          null
-      }
-      <h1 className='stats__heading'>{props.title}</h1>
-      <div className='stats__content'>
-        {
-          props.stats.map(stat => (
-            <div className='stats__stat' key={stat.name + stat.value}>
-              <div className='stats__value'>{stat.value}</div>
-              <div className='stats__statName'>{stat.name}</div>
-            </div>
-          ))
-        }
-      </div>
-    </section>
-  );
-};
+  /**
+   * Stats component class constructor
+   * @param {*} props - title, stats and page information
+   */
+  constructor(props) {
+    super(props);
+    const { title, stats, page } = this.props;
+    this.state = {
+      title,
+      stats,
+      page,
+    };
+  }
 
-Stats.defaultProps = {
-  title: '',
-  page: '',
-};
+  /**
+   * Render Stats Component
+   * @return {object} JSX for Stats Component
+   */
+  render() {
+    const { title, page, stats } = this.state;
+    /**
+     * @name icons
+     * Society icons
+     */
+    const icons = {
+      invictus: InvictusIcon,
+      istelle: IstelleIcon,
+      phoenix: PhoenixIcon,
+      sparks: SparksIcon,
+    };
 
-/**
-    * @name propTypes
-    * @type {PropType}
-    * @param {Object} propTypes - React PropTypes
-    * @property {Object[]} stats - Statistics to display
-  */
-Stats.propTypes = {
-  title: PropTypes.string,
-  page: PropTypes.string,
-  stats: PropTypes.arrayOf(PropTypes.shape({
-    name: PropTypes.string.isRequired,
-    value: PropTypes.string.isRequired,
-  })).isRequired,
-};
+    const SocietyIcon = title && icons[title.toLowerCase()];
+    /**
+     * @function hasIcon
+     * @returns {bool} whether society icon is in icons object
+     */
+    const hasIcon = () => (
+      Object.keys(icons).find(statsTitle => statsTitle === title.toLowerCase())
+    );
+
+    return (
+      <section className='stats'>
+        {
+          page === 'society' ?
+            <span className='sidebar__navIcon sidebar__navIcon--black'>
+              {hasIcon() ? <SocietyIcon /> : null}
+            </span>
+            :
+            null
```

Comment:
Airbnb style guidelines forbids nesting ternaries

Reply:
This isn't quite the case.  The `no-nested-ternary` rule is to enhance readability. What we have here is an if statement in an if block, not an if ... else if situation hence the code is very much readable. Had the code been as such: `page === 'society' ? hasIcon() ? <SocietyIcon /> : null : null` then I would concur. (https://eslint.org/docs/rules/no-nested-ternary).

Reply:
We have extended the airbnb rules so eslint would have caught this. To verify paste this `let thing = foo ? bar : baz === qux ? quxx : foobar` into your code, eslint should give you the `Do not nest ternary expressions. (no-nested-ternary)` Warning